### PR TITLE
MAID-2413: S3 deploy

### PR DESCRIPTION
**Pull request by hunterlester**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+dist: trusty
+os:
+- linux
+language: node_js
+node_js:
+- '7'
+install:
+- node --version
+- npm --version
+- npm i
```
Comment (joshuef) [inline .travis.yml]: Let's use `yarn` over `npm` as we're using that in the projects in general. 

@hunterlester you'll need to add yarn cache to the setup of the image also.

**Review by joshuef — CHANGES_REQUESTED** [review]

File: src/api/ipc.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ const reqsSent = new Map();
 
 let ipcEvent = null;
 // Until ipcEvent is ready, we need to keep reqs in a queue
-var pendingReqs = [];
+const pendingReqs = [];
```
Comment (bochaco) [inline src/api/ipc.js:10]: I don't think this should be const as it needs to be modified afterwards

File: src/api/ipc.js [hunk 3]
```diff
@@ -20,35 +20,29 @@ class AuthRequest {
...
+    // let's keep it in a queue to send when ipcEvent is ready
+    pendingReqs.push(req);
   }
+}
 
-  remove(id) {
-    reqsSent.delete(id);
-    return this;
-  }
+function sendPendingReqs() {
```
Comment (bochaco) [inline src/api/ipc.js]: Let's use arrow functions to have a consistency.

**Review by bochaco — CHANGES_REQUESTED** [review]

File: src/api/ipc.js (hunk 2 above)
Comment (hunterlester) [reply src/api/ipc.js:10]: Linter rule is that the value is never reassigned. Looks like `const` is fine here.

File: src/api/ipc.js [hunk 4]
```diff
@@ -20,44 +20,38 @@ class AuthRequest {
...
-  add(uri, isUnregistered, cb) {
-    let req = new AuthRequest(uri, isUnregistered, cb);
-    if (ipcEvent) {
-      reqsSent.set(req.id, req);
-      ipcEvent.sender.send('webClientAuthReq', req);
-    } else {
-      // let's keep it in a queue to send when ipcEvent is ready
-      pendingReqs.push(req);
-    }
+function add(uri, isUnregistered, cb) {
```
Comment (bochaco) [inline src/api/ipc.js]: Let's use arrow function here also for consistency.

File: src/api/ipc.js [hunk 5]
```diff
@@ -20,44 +20,38 @@ class AuthRequest {
...
+    ipcEvent.sender.send('webClientAuthReq', req);
+  });
+  pendingReqs.length = 0;
+};
 
-  remove(id) {
-    reqsSent.delete(id);
-    return this;
-  }
+function remove(id) {
```
Comment (bochaco) [inline src/api/ipc.js]: Let's use arrow function here also for consistency.

**Review by bochaco — CHANGES_REQUESTED** [review]

**Review by bochaco — APPROVED** [review]

**Pull request merged**